datadog/resource_datadog_role.go

```diff
@@ -0,0 +1,287 @@
+package datadog
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/DataDog/datadog-api-client-go/api/v2/datadog"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/customdiff"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+// validPermissions is a map of all unrestricted permission IDs to their name
+var validPermissions map[string]string
+
+func resourceDatadogRole() *schema.Resource {
+	return &schema.Resource{
+		Exists: resourceDatadogRoleExists,
+		Create: resourceDatadogRoleCreate,
+		Read:   resourceDatadogRoleRead,
+		Update: resourceDatadogRoleUpdate,
+		Delete: resourceDatadogRoleDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceDatadogRoleImport,
+		},
+		CustomizeDiff: customdiff.ValidateValue("permission", validatePermissionsUnrestricted),
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of the role.",
+			},
+			"permission": {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Description: "List of permission IDs to give to this role.",
```

Comment:
is it list of permission IDs or objects with id and name?

Reply:
It is a list of objects with IDs and name (name being computed so user only gives the IDs). I'm not using the name in the provider code directly anymore (needed it when I was exploring the suppressDiff approach), but I figured I'd leave it like this as I thought it might be useful for users to reference maybe.
I'm fine going back to a pure list of ID strings only, if you think it makes more sense

Reply:
I would only update the description to match the new schema.

Reply:
```suggestion
				Description: "List of objects containing the permission ID the name of the permissions assigned to this role.
```